clang/lib/CConv/ConstraintBuilder.cpp

```diff
@@ -34,9 +36,7 @@ void processRecordDecl(RecordDecl *Declaration, ProgramInfo &Info,
         // We only want to re-write a record if it contains
         // any pointer types, to include array types.
         for (const auto &D : Definition->fields()) {
-          Info.getABoundsInfo().insertVariable(D);
           if (D->getType()->isPointerType() || D->getType()->isArrayType()) {
-            Info.addVariable(D, Context);
```

Comment:
Why are these getting removed?

Reply:
I see now -- you have a separate visitor for adding variables. That's a nice idea. I suppose that if the tests pass we can assume you got every case. :)